### PR TITLE
Extend email plugin settings api

### PR DESCRIPTION
**Pull request by bernardjkim**
This PR extends the plugin spec API to support Email Plugin settings.
This changes is required to support hosted Email plugins.

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6576,6 +6578,42 @@ message AWSICProvisioningSpec {
...
+
+// SMTPSpec holds a generic SMTP service specific settings.
+message SMTPSpec {
+  option (gogoproto.equal) = true;
+
+  // Host specifies the SMTP service host name.
+  string host = 1;
+  // Port specifies the SMTP service port number.
+  int32 port = 2;
+}
```
Comment (hugoShaka) [inline api/proto/teleport/legacy/types/types.proto:6615]: Where do we store the starttls setting of the email plugin?

**Review by hugoShaka — APPROVED** [review]
LGTM, do we need to block the SMTP mode of the plugin in cloud to avoid being used to probe SMTP servers or send spam? (here is not the right PR for this, if we block SMTP on cloud this should happen in the next teleport.e PR)

**Review by marcoandredinis — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (bernardjkim) [reply api/proto/teleport/legacy/types/types.proto]: Ah I missed this config. Thanks for pointing it out. I've added the additional config option to the api.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -6612,6 +6612,27 @@ message SMTPSpec {
...
+  SMTP_MANDATORY_START_TLS = 0;
+
+  // SMTP_OPPORTUNISTIC_START_TLS means that SMTP transactions are encrypted if
+  // STARTTLS is supported by the SMTP server. Otherwise, messages are
+  // sent in the clear.
+  SMTP_OPPORTUNISTIC_START_TLS = 1;
+
+  // SMTP_NO_START_TLS means encryption is disabled and messages are sent in the
+  // clear.
+  SMTP_NO_START_TLS = 2;
```
Comment (hugoShaka) [inline api/proto/teleport/legacy/types/types.proto:6635]: Usually we leave the 0 enum case for the UNSPECIFIED case.

Also enum cases must start with the enum name I think, else the buf linter will not let your PR pass

**Comment by hugoShaka** [comment]
> That's a valid concern. I'll omit the SMTP option in the WebUI.

Even if it's not in the web UI, you can create the plugin manually in the CLI, the backend service will need to reject those if `modules.IsCloud()`

**Pull request merged**